### DISCUSSION TITLE
Blazor traffic parsing

### DISCUSSION THREAD

**Discussion by driuba** [mitmproxy/mitmproxy#6610] [Ideas]
Recently I encountered a need to work with Blazor and I noticed that there is no option to parse its traffic (`WebSocket`, `LongPolling`, etc.).

As far as I can tell the communication is based on SignalR message pack serialization. Would it be useful, practical to implement some sort of parsing for protocol?

I've seen some examples in the form or Burp extension, that parses blazorpack into JSON and back for preview and editing. I've also seen some parsing examples in Python. However, I haven't tested any of them.

How realistic is such idea and if I were to go about implementing it, where should I start?

**Comment by ssstonebraker** [comment]
@driuba  check this out - https://sensepost.com/blog/2023/decoding-blazorpack/
